source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -851,6 +851,20 @@ ClientContextImpl::ClientContextImpl(Stats::Scope& scope,
   }
 }
 
+bool ContextImpl::parseAndSetAlpn(const std::vector<std::string>& alpn, SSL& ssl) {
+  if (alpn.empty()) {
+    return false;
+  }
+  std::vector<uint8_t> parsed_alpn = parseAlpnProtocols(absl::StrJoin(alpn, ","));
+  if (!parsed_alpn.empty()) {
+    const int rc = SSL_set_alpn_protos(&ssl, parsed_alpn.data(), parsed_alpn.size());
+    RELEASE_ASSERT(rc == 0, Utility::getLastCryptoError().value_or(""));
```

Comment:
Maybe a comment on when this fails (is it only an OOM scenario?)

Reply:
Yeah that's what it looks like:

```
int SSL_set_alpn_protos(SSL *ssl, const unsigned char *protos,
                        unsigned int protos_len)
{
    OPENSSL_free(ssl->ext.alpn);
    ssl->ext.alpn = OPENSSL_memdup(protos, protos_len);
    if (ssl->ext.alpn == NULL) {
        SSLerr(SSL_F_SSL_SET_ALPN_PROTOS, ERR_R_MALLOC_FAILURE);
        return 1;
    }
    ssl->ext.alpn_len = protos_len;

    return 0;
}
```

I'll add a comment